common/defaults.go

```diff
@@ -662,6 +653,33 @@ func (r *RedisConnectorConfig) SetDefaults() error {
 		r.SetTimeout = Duration(2 * time.Second)
 	}
 
+	// URI is provided directly
+	if r.URI != "" {
+		return nil
+	}
+
+	// URI needs to be constructed from individual fields
+
+	// Clean up Addr if it has scheme prefixes
+	r.Addr = strings.TrimPrefix(r.Addr, "rediss://")
+	r.Addr = strings.TrimPrefix(r.Addr, "redis://")
+
+	// Construct URI from discrete fields
+	if r.Addr != "" {
+		var userInfo string
+		if r.Username != "" || r.Password != "" {
+			userInfo = r.Username + ":" + r.Password + "@"
+		}
```

Comment:
```suggestion
		if r.Password != "" {
			userInfo = r.Username + ":" + r.Password + "@"
		}
```
Probably this is what we need, for cases where username is not provided and looks like this URI is valid: `redis://:mypassword@host`